designs/tinkerbell-provider-input.md

```diff
@@ -0,0 +1,366 @@
+* Push the hardware directly to the Tinkerbell stack so users don't have to do it manually
+
+## Solution details
+
+### Cluster Configuration 
+
+The first file users will need is the clusterconfig file, which they can use to configure their EKS Anywhere cluster.
+
+To generate the clusterconfig for Tinkerbell provider, we use the existing `generate` command and specify `tinkerbell` as the provider
+```bash
+eksctl anywhere generate clusterconfig <cluster-name> -p tinkerbell
+```
+
+For Tinkerbell provider, we have introduced two new CRDs:
+* TinkerbellDatacenterConfig
+* TinkerbellMachineConfig
+
+**TinkerbellDatacenterConfig**
+
+TinkerbellDatacenterConfig describes the configuration details for Tinkerbell stack.
+
+In this config, users must specify the following fields:
+* spec.tinkerbellIP: defines the IP of the machine running the Tinkerbell stack.
+* spec.tinkerbellCertURL: defines the endpoint where the Tinkerbell stack hosts its certificate. This is used to communicate with the stack.
+* spec.tinkerbellGRPCAuth: defines the GRPC endpoint used by Tinkerbell stack. This is also used to communicate with the stack.
+* spec.tinkerbellPBnJGRPCAuth: defines the GRPC endpoint used by PNBJ for hardware remote management. This is used to change the power state and the boot order of the servers.
+
+Here's an example `TinkerbellDatacenterConfig`
+
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: TinkerbellDatacenterConfig
+metadata:
+  name: eks-a-tinkerbell-cluster
+spec:
+  tinkerbellIP: "192.168.0.20"
+  tinkerbellCertURL: "http://192.168.0.20:42114/cert"
+  tinkerbellGRPCAuth: "192.168.0.20:42113"
+  tinkerbellPBnJGRPCAuth: "192.168.0.20:50051"
+```
+
+**TinkerbellMachineConfig**
+
+TinkerbellMachineConfig describes the configuration for machines used for creating EKS Anywhere cluster on Tinkerbell.
+
+This config supports the following fields
+* spec.osFamily (required): defines the operating system to be installed on this machine. We currently only support ubuntu for this but this list is expected to grow in the future.
+* spec.users (required): defines the users to be created on this machine along with the ssh keys.
+  * spec.users[].name (required): defines the name of the user.
+  * spec.users[].sshAuthorizedKeys (required): defines the SSH authorized keys to be installed on this machine to be used for SSH'ing into it.
+* spec.templateOverride (optional): defines the [Tinkerbell template](https://docs.tinkerbell.org/templates/) to be used to provision the OS on the machine. This field is passed as a YAML encoded as string.
+
+Here is an example `TinkerbellMachineConfig`
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: TinkerbellMachineConfig
+metadata:
+  name: eks-a-tinkerbell-cluster-cp
+spec:
+  osFamily: ubuntu
+  users:
+  - name: ec2-user
+    sshAuthorizedKeys:
+    - ssh-rsa AAAA...
+  templateOverride: |
+    <>
+```
+
+Here is the default `templateOverride` that is generated in the clusterconfig
```

Comment:
So the `templateOverride` gets populated in `generate clusterconfig` or do these values get filled in from datacenter and machine configs?

Reply:
It gets generated when you run `generate clusterconfig`